transaction/atomic.go

```diff
@@ -0,0 +1,148 @@
+package transaction
+
+import (
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+
+	"github.com/hermeznetwork/hermez-go-sdk/account"
+	"github.com/hermeznetwork/hermez-go-sdk/client"
+	"github.com/hermeznetwork/hermez-node/api"
+
+	ethCommon "github.com/ethereum/go-ethereum/common"
+	hezcommon "github.com/hermeznetwork/hermez-node/common"
+)
+
+type AtomicTxItem struct {
+	SenderBjjWallet       account.BJJWallet
+	RecipientAddress      string
+	TokenSymbolToTransfer string
+	Amount                *big.Int
+	FeeRangeSelectedID    int
+	RqOffSet              int
+}
+
+func getAccountDetails(hezClient client.HermezClient, address string,
+	tokenToTransfer string) (idx hezcommon.Idx, nonce hezcommon.Nonce, tokenId hezcommon.TokenID, err error) {
+	var accDetails account.AccountAPIResponse
+	accDetails, err = account.GetAccountInfo(hezClient, address)
+	if err != nil {
+		err = fmt.Errorf("error obtaining account details. Account: %s - Error: %s\n", address, err.Error())
+		return
+	}
+	for _, innerAccount := range accDetails.Accounts {
+		if strings.ToUpper(innerAccount.Token.Symbol) == tokenToTransfer {
+			tokenId = hezcommon.TokenID(innerAccount.Token.ID)
+			nonce = hezcommon.Nonce(innerAccount.Nonce)
+			tempAccountsIdx := strings.Split(innerAccount.AccountIndex, ":")
+			if len(tempAccountsIdx) == 3 {
+				var tempAccIdx int
+				tempAccIdx, err = strconv.Atoi(tempAccountsIdx[2])
+				if err != nil {
+					err = fmt.Errorf("error getting account index. Account: %+v - Error: %s\n", innerAccount, err.Error())
+					return
+				}
+				idx = hezcommon.Idx(tempAccIdx)
+			}
+		}
+	}
+	if len(strconv.FormatUint(uint64(idx), 10)) < 1 {
+		err = fmt.Errorf("there is no account to this user %s for this Token %s", address, tokenToTransfer)
+		return
+	}
+	return
+}
+
+// AtomicTransfer perform token or ETH transfers in a pool of transactions
+func AtomicTransfer(hezClient client.HermezClient, ethereumChainID int,
```

Comment:
This function should receive the txs already signed, as the common case will be for no single entity to have all the related private keys to an atomic group. So this function should take care of:

- Set the `RqOffset`
- Set the `AtomicGroupID`
- Perform the HTTP POST

Of course all of this already happens, just need to change the interface of the function to have as input `txs []common.PoolL2Tx`. 

Additionally, it would be nice to validate that:

- Txs are valid (signature is correct)
- The given set of transactions represent a single atomic group

This is actually same validations that the `hermez-node` will run. But I believe running the validations before sending the request may be easier to debug, and can give more info to users.

Keep in mind that the current code is actually working (I believe), so it's up to you to implement this or not, in this PR or in future ones. The only thing I would say it's mandatory is to don't sign the txs in this function, as this will make the SDK unusable

Reply:
This will be fixed at https://github.com/hermeznetwork/hermez-go-sdk/issues/4